pkg/kubelet/kubelet_node_status.go

```diff
@@ -479,7 +480,7 @@ func (kl *Kubelet) setNodeAddress(node *v1.Node) error {
 			var addrs []net.IP
 			addrs, err = net.LookupIP(node.Name)
 			for _, addr := range addrs {
-				if !addr.IsLoopback() && addr.To4() != nil {
+				if err = kl.validateNodeIP(); err == nil && (addr.To4() != nil || addr.To16() != nil) {
```

Comment:
It is a bit strange, on line 481 net.LoopIP returns addrs and err, but then inside of for loop its value will be lost as the same err variable is used. Maybe on line 481 instead of addrs, err should be used addrs, _
or on line 483 if err := kl.blah blah. Apologies if it was done intentionally and I just did not get the design.

Reply:
@danehans Was the intent to ignore any errors from LookupIP() call on L488?

Also, if there are addresses for the host, would they ever be invalid? Wondering if we just need to check if not LB, multicast, etc. as tested by validateNodeIP().

Lastly, should we be giving precedence to any IPv4 address over IPv6 address or can either be returned?

Reply:
I will update the block to handle the err from `LookupIP` before the loop.

@pmichali we will want to avoid using some addresses on the host such as loopback, etc.. The loop will select the first usable address returned by the validateNodeIP method for the `v1.NodeInternalIP`.

I will update so v4 is preferred over v6.